### DISCUSSION TITLE
Giving users only access to read their table

### DISCUSSION THREAD

**Discussion by HosAkh** [supabase/supabase#1639] [Questions] [answered]
I have a list of tables that I'm accessing via logging into the database and visualizing but I want to give other users access to only read their own table via api. So they can read their data via api but that's it. 

I invited the user and it's validated
click `add policy`
click `select` 
gave it a name
definition: auth.email='[usersEmailHere]' <----there is an actual email address just to clarify
but I get error code: `missing FROM-clause entry for table “auth”`

sorry about the noob question but I'm stuck and don't understand how to only allow users to be able to access their own table via API only.

**Comment by easy-street** [comment, marked as answer]
this error is caused by missing parens in the definition. example policy definition using the `auth.email` function with a hard-coded email address:
```
auth.email() = 'email@example.com'
```

**Reply by HosAkh** [reply]
it gave me the error of `function auth.email() does not exist`. 

edit: i'm confused by why it says that as I thought `auth.email()` was a sql lang thing.

**Reply by easy-street** [reply]
i noticed that as well. you can find instructions to create the function manually in [this comment](https://github.com/supabase/supabase/discussions/1147#discussioncomment-608722) and track this issue via the discussion there.

**Reply by HosAkh** [reply]
I've been lost all day! This worked, thank you!
